### DISCUSSION TITLE
Use with `rmagatti/auto-session`?

### DISCUSSION THREAD

**Discussion by bennypowers** [nvim-neo-tree/neo-tree.nvim#126] [Q&A] [answered]
Hello!

I'm using neo-tree with `auto-session` and `bufferline.nvim`, and notice the following:

- WHEN neo-tree is open and I quit vim with buffers open in session via `quitall`
- THEN when I reopen the session from the shell prompt, 
  - The left pane (where the tree should be) is blank
  - A new, empty buffer appears in the buffer line labeled `neo-tree filesystem [1]`
  - the bufferline, normally offset via bufferline config, is flush to the left of the screen

<img width="1014" alt="Screen Shot 2022-02-06 at 11 06 34" src="https://user-images.githubusercontent.com/1466420/152674368-7d6b3567-66de-4e05-969a-f08da25e8ff4.png">
<img width="1014" alt="Screen Shot 2022-02-06 at 11 06 40" src="https://user-images.githubusercontent.com/1466420/152674370-ab26ee75-0594-49ed-b4c6-855a336536af.png">

I'd like instead for the session to restore as is, i.e.

- WHEN neo-tree is open and I quit vim with buffers open in session via `quitall`
- THEN when I reopen the session from the shell prompt, 
  - The left pane (where the tree should be) contains a neo-tree with the same state as at quit
  - the bufferline offsets from the tree as it did before quit
  - the state of the bufferline persists

Should I hook into auto-session to prevent saving buffers with type `neo-tree` and to focus the active buffer on startup?

**Comment by cseickel** [comment, marked as answer]
I have two answers for this. The first is that I use use Startify for sessions, and I have always had a "before save" function that closes anything where the filetype has `"tree"` in the name because this is a universal problem:

```
function! CustomStartifyBeforeSave() abort
    for win in nvim_list_wins()
        " close floating windows
        if nvim_win_get_config(win).relative > ""
            call nvim_win_close(win, 1)
        else
            " close drawer and tool windows
            let l:ft = nvim_buf_get_option(nvim_win_get_buf(win), "ft")
            if l:ft =~ "tree" || l:ft == "fern" || l:ft == "Trouble"
                call nvim_win_close(win, 1)
            endif
        endif
    endfor
endfunction
let g:startify_session_before_save = ['call CustomStartifyBeforeSave()']
```

The right thing to do right now is to implement similar logic for auto-session.
 
In the future however, it would be pretty awesome if Neo-tree could restore itself when the session is restored. I am going to put that on the roadmap.
